pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2008,128 +2162,210 @@ static __always_inline int do_should_trace(program_data_t *p)
         return 0;
     }
 
-    if ((config & FILTER_FOLLOW_ENABLED) && (proc_info->follow)) {
-        // don't check the other filters if follow is set
-        return 1;
-    }
+    if (p->config->cont_filter_enabled_scopes) {
+        // e.g.: cmdline: -t 5:container
+        //
+        // filter_out_scopes             =  0000 0000
+        // cont_filter_enabled_scopes    =  0001 0000 (scope 5)
+        // mask                          = ~0001 0000 = 1110 1111
+        // res                           =  1111 1111 <- ~0ULL
+        //
+        // ---
+        //
+        // considering an event from a container
+        //
+        // bool_filter_matches(0, true)  = 1111 1111
+        //
+        // res                           = 1111 1111 &
+        //                                 1111 1111 |
+        //                                 1110 1111
+        //                                 ---------
+        //                                 1111 1111 (scope 5 matched)
+        //
+        // considering an event not from a container
+        //
+        // bool_filter_matches(0, false) = 0000 0000
+        //
+        // res                           = 1111 1111 &
+        //                                 0000 0000 |
+        //                                 1110 1111
+        //                                 ---------
+        //                                 1110 1111 (scope 5 unmatched)
```

Comment:
I think this comment is too much and repeats itself.
The comment should mostly explain about mask, which we didn't explain about in the filtering functions. Something like:
`For scopes which has this filter disabled we want to set the matching bits using 'mask'`
I would also put this comment right before computing `res` below

Reply:
@rafaeldtinoco @NDStrahilevitz @josedonizetti I would appreciate your reply on this matter, since you're not so acquainted to the multiscope logic.

Reply:
I'm okay with removing the redundant comments. I do think having a full example is very important, though.